element_facemap/facial_behavior_estimation.py

```diff
@@ -0,0 +1,361 @@
+        "The argument 'dependency' must be a module's name or a module"
+    assert hasattr(linking_module, 'get_facemap_root_data_dir'),\
+        "The linking module must specify a lookup function for a root data directory"
+
+    global _linking_module
+    _linking_module = linking_module
+
+    # activate
+    schema.activate(facemap_schema_name, create_schema=create_schema,
+                    create_tables=create_tables,
+                    add_objects=_linking_module.__dict__)
+
+
+# -------------- Functions required by element-facemap ---------------
+
+def get_facemap_root_data_dir() -> list:
+    """
+    It is recommended that all paths in DataJoint Elements stored as relative
+    paths, with respect to some user-configured "root" director(y/ies). The
+    root(s) may vary between data modalities and user machines
+    get_fm_root_data_dir() -> list
+        This user-provided function retrieves the possible root data
+        director(y/ies) containing continuous face data for all subjects
+        and sessions (e.g. acquired video raw files)
+        :return: a string for full path to the behavioral root data directory,
+         or list of strings for possible root data directories
+    """
+    root_directories = _linking_module.get_facemap_root_data_dir()
+    if isinstance(root_directories, (str, Path)):
+        root_directories = [root_directories]
+
+    if hasattr(_linking_module, 'get_facemap_processed_data_dir'):
+        root_directories.append(_linking_module.get_facemap_processed_data_dir(None))
+
+    return root_directories
+
+
+def get_facemap_processed_data_dir() -> str:
+    """
+    If specified by the user, this function provides Facemapp with an output
+    directory for processed files. If unspecified, output files will be stored
+    in the session directory 'videos' folder, per Facemap default
+    get_fm_processed_data_dir -> str
+        This user-provided function specifies where Facemap output files
+        will be stored.
+    """
+    if hasattr(_linking_module, 'get_facemap_processed_data_dir'):
+        return _linking_module.get_facemap_processed_data_dir()
+    else:
+        return get_facemap_root_data_dir()[0]
+
+
+def get_facemap_video_files(video_key: dict) -> str:
+    """
+    Retrieve the list of video files (e.g. *.avi) associated with a given video recording
+    :param video_key: key of a video recording
+    :return: list of Video files' full file-paths
+    """
+    return _linking_module.get_facemap_video_files(video_key)
+
+
+# ----------------------------- Table declarations ----------------------
+
+
+@schema
+class VideoRecording(dj.Manual):
+    definition = """
+    -> Session
+    -> Device
+    recording_id                : int
+    """
+
+    # One VideoRecording can be saved in multiple files
+    class File(dj.Part):
+        definition = """
+        -> master
+        file_path: varchar(255)     # filepath of video, relative to root data directory
```

Comment:
do not include the filepath in the primary key.  You can number of the files but the path should be a secondary attribute

Reply:
Definitely! Changed to:
```python
    class File(dj.Part):
        definition = """
        -> master
        file_id         : int           # file id
        ---
        file_path       : varchar(255)  # filepath of video, relative to root data directory
        """
```